modules/programs/senpai.nix

```
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.programs.senpai;
+in {
+  options.programs.senpai = {
+    enable = mkEnableOption "senpai";
+    config = mkOption {
+      type = types.attrs;
+      example = {
+        addr = "example.org:6697";
+        nick = "yournick";
+      };
```

```suggestion
with lib;

let
  cfg = config.programs.senpai;
  cfgType = pkgs.formats.yaml { };
in
{
  options.programs.senpai = {
    enable = mkEnableOption "senpai";
    config = mkOption {
      type = cfgType.type;
      example = literalExample ''
        {
          addr = "example.org:6697";
          nick = "yournick";
        }
      '';
```

Please use a structed type.

If possible, could you use a more refined type that would allow some configuration parameters to be typed.

``` nix
config = mkOption {
  type = types.submodule {
    freeformType = cfgType.type;
    options = {
      
    };
  };
}
```

Using what is called a freeform module allows you to define some options and leave the rest undeclared.

This improves documentation and reduces the burden of keeping options up to date.

We usually only add options for settings that are always used.